src/server/game/Scripting/ScriptMgr.h

```diff
@@ -194,6 +194,13 @@ class WorldScript : public ScriptObject
 
     // Called when the world is actually shut down.
     virtual void OnShutdown() { }
+
+    /**
+     * @brief This hook runs before finalizing the player world session. Can be also used to mutate the cache version of the Client.
+     *
+     * @param version The cache version that we will be sending to the Client.
+     */
+    virtual void OnBeforeFinalizePlayerWorldSession(uint32& /*version*/) {}
```

Comment:
```suggestion
    virtual void OnBeforeFinalizePlayerWorldSession(uint32& /*cacheVersion*/) {}
```

Maybe this is a better param name?

Reply:
@Kitzunu Agreed and pushed.